### ISSUE TITLE
[Feature Request] ShowToast can return handle to toast instance in order to close or update it.

### ISSUE THREAD

**Issue by Liero** [Blazored/Toast#97]
## SCENARIO
I would like to show toasts instead of wait overlays in different scenarios.

For example when user click a button, I'm sending some emails in the background and during the process I would like to show a toast with messing "Sending emails..". When done sending, I would like to either close the toast, or update it to "Emails sent".

var toastReference = await toastService.ShowToast(ToastLevel.Info, "Sending emails");
await Task.Delay(1000);
toastReference.Timeout = TimeSpan.MaxValue; //or TimeSpan.Zero or null?
toastReference.Level = ToastLevel.Success;
toastReference.Message = "Emails sent!";
toastReference.ResetTimeout(TimeSpan.FromSeconds(2));
toastReference.Close();

**Comment by chrissainty** [comment]
Thanks for raising this @Liero. It is actually already being worked on as part of the next major version! I'll tag this issue as part of that milestone so we make sure we don't overlook anything.

**Comment by Liero** [comment]
Hi @chrissainty,
is there any progress on this?

I'm sure you are quite busy, so let me know if you need help.

I need customizable timeout per toast instance. Is it realistic, it will be released this year? Otherwise I will need to fork the repo.

**Comment by chrissainty** [comment]
I'm just in the last stages with my book and I'm just getting it finalised for the publishers. Once that is done, I'll have some time back again and I can start getting some of these issues sorted. If you can just bare with me a little longer.

**Comment by adamashton** [comment]
I am looking to close a toast automatically once a long running task has completed. It would be great to reference a toast/id a close the specific one via the ToastService.

**Comment by Cvijo** [comment]
What about if ShowToast methods return ToastInstance class (need to change to public class), set all properties to internal except for maybe `Guid`, and create a new `public void Close()` method

something like this:

<pre>
<code>
public class ToastInstance
{
    <i><b>internal event Action<Guid>? OnClose;  </b></i>
<h6>
    internal ToastInstance(RenderFragment message, ToastLevel level, ToastSettings toastSettings)
    {
        Message = message;
        Level = level;
        ToastSettings = toastSettings;
    }
    internal ToastInstance(RenderFragment customComponent, ToastSettings settings)
    {
        CustomComponent = customComponent;
        ToastSettings = settings;
    }

    public Guid Id { get; } = Guid.NewGuid();
    internal DateTime TimeStamp { get; } = DateTime.Now;
    internal RenderFragment? Message { get; }
    internal ToastLevel Level { get; }
    internal ToastSettings ToastSettings { get; }
    internal RenderFragment? CustomComponent { get; }
</h6>
<i><b> 
    public void Close()
    {
        OnClose?.Invoke(Id);
    }
 </b></i>
}

</code>
</pre>

inside `BlazoredToasts.cs` on `ShowToast`

<pre><code>
    private ToastInstance ShowToast(ToastLevel level, RenderFragment message, Action<ToastSettings>? toastSettings)
    {
<b>
        var settings = BuildToastSettings(level, message, toastSettings);
        var toast = new ToastInstance(message, level, settings);
        toast.OnClose += RemoveToast; </b>
        InvokeAsync(() =>
        {

            if (ToastList.Count < MaxToastCount)
            {
                ToastList.Add(toast);
                StateHasChanged();
            }
            else
            {
                ToastWaitingQueue.Enqueue(toast);
            }
        });
        <b>return toast; </b>
    }
</code></pre>


using it would be something like:

```
        var instance = toastService.ShowToast(ToastLevel.Warning, "message to toast", settings => settings.DisableTimeout= true);
        await Task.Delay(2000); //<-- waiting for some task to complete
        instance.Close();

```

Another thing maybe to consider is to have something like an instance.SetMessage() to be able to change the message, for example, if you are waiting some multiple task and you show a toast with DisableTimeout = true and to update message "task 1 done" .. "task 2 done", "last task in progress" and then instance.Close() might be useful in some cases.

**Comment by chrissainty** [comment]
@Cvijo I think this could be a good starting point. Do you want to raise a PR based on your suggestion and then we can see how it looks in reality?

**Comment by Liero** [comment]
I've already done it, see [ToastService.cs](https://github.com/Liero/vNext.BlazorComponents.Toasts/blob/d7468c2acfbe7db46ad849d8f60187b231758359/vNext.BlazorComponents.Toasts/ToastService.cs#L30) and [ToastReference.cs](
https://github.com/Liero/vNext.BlazorComponents.Toasts/blob/main/vNext.BlazorComponents.Toasts/ToastReference.cs)

I use the same approach for updating the toast (e.g. I change title, content, severity etc of existing toast)
